lib/mongo/server/connection_pool/queue.rb

```diff
@@ -88,7 +87,7 @@ def disconnect!
         # @since 2.0.0
         def enqueue(connection)
           mutex.synchronize do
-            queue.unshift(connection)
+            queue.unshift(connection.record_checkin!)
             resource.broadcast
           end
         end
```

Comment:
The Ruby driver doesn't connect the Connection objects until they are checked out of the pool the first time. So they wouldn't need to be disconnected until they are checked out and checked back in (and at that point, they would have a last_checkin value). Does that make sense?

[Here](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/server/connection.rb#L135) is where the Connection object is initialized.
And then, when it is used for the first time, [this method](https://github.com/mongodb/mongo-ruby-driver/blob/master/lib/mongo/server/connectable.rb#L93-L103) is called that connects it and creates a socket for the Connection.